components/infobox/commons/infobox_team.lua

```diff
@@ -220,8 +220,12 @@ function Team:processCreateDates()
 		local cleanDate = ReferenceCleaner.clean(date)
 
 		if game:lower() == 'org' then
-			icon = self:_getTeamIcon(cleanDate)
-			icon = String.isNotEmpty(icon) and '[[File:' .. icon .. ']]' or nil
+			local teamIcon, teamIconDark = self:_getTeamIcon(cleanDate)
+			local toIcon = function(iconFile, mode)
+				if String.isEmpty(iconFile) then return '' end
+				return '[[File:' .. iconFile .. '|class=show-when-' .. mode .. '-mode]]'
+			end
+			icon = toIcon(teamIcon, 'light') .. toIcon(teamIconDark, 'dark')
```

Comment:
I think extracting this into a standard lib is a good idea. Should probably support so when one of the two is missing, it would not add the show-when class.

Reply:
you mean specific for team icons or just generally for files?
also what should we name it?

Reply:
#3996